### ISSUE TITLE
Login Image not appearing after install

### ISSUE THREAD

**Issue by Shadowstreik** [nextcloud/nextcloudpi#35]
Hello. I have installed NextCloudPi based on the OwnYourBits instruction. Everything runs great. However, the login image does not appear. Instead, only a solid background color appears. This applies to the default login image and any other image file I select to serve as that image.

I have run the LetsEncrypt process, but am not certain if that would have any effect.

Would there be a permissions issue at play, here?

**Comment by Shadowstreik** [comment]
Navigate to...

> /var/www/nextcloud/config

enter the config file...

> sudo nano config.php

Find the trusted domains...

```
  'trusted_domains' => 
  array (
    0 => 'localhost',
    1 => '192.168.xxx.xxx',
    4 => 'letsencrypt domain',
```

Add the domain of the actual IP the subdomain points to...

>     2 => 'actual.domain.letsencrypt.points.to',

Save file and exit from Nano.

Do not simply restart the Apache webservice, but a warm reboot will suffice.

Issue is resolved. Closing issue.

**Comment by nachoparker** [comment]
Hi,
I am glad you got it working.

I do not understand very well... isn't it `2 => actual.domain` the same as ` 4 => letsencrypt.domain`? or am I missing some configuration that you have?

Also, I am interested if anybody else run into this.

Thanks for your feedback

**Comment by nachoparker** [comment]
I'll reopen this until we clarify. Maybe there is something else that can be done from NCP to avoid this situation.

**Comment by Shadowstreik** [comment]
>  isn't it 2 => actual.domain the same as 4 => letsencrypt.domain?

I would think it is, or should be, just by how I understand it. Though, I would enter the letsencrypt.domain and be met with the solid background color, only. It wasn't until I did an entry with actual.domain that the logo and login image would appear. I would agree that the letsencrypt.domain should be sufficient.

I also notice during my syncs that the same issue would be encountered for larger file syncs. But, I think that just may be a performance limitation by the RPi3, itself.

**Comment by Shadowstreik** [comment]
Update- I just went to the letsencyrpt.domain and the login background image and logo image are not there. It is only the background color. I own the domain I'm using for the letsencrypt.domain, not that that would make any difference/"should".

**Comment by Shadowstreik** [comment]
![screenshot from 2017-07-04 14-37-00](https://user-images.githubusercontent.com/29790545/27842822-52c3e318-60db-11e7-9786-159d3c5faa7d.png)

**Comment by Shadowstreik** [comment]
Update- I shut off Fail2Ban and the images load, now. I'm going to watch it the next couple of days. If it still shows, Fail2Ban may be the problem.

**Comment by nachoparker** [comment]
fail2ban... doesn't sound likely.

Do you have `modsecurity` on?

**Comment by nachoparker** [comment]
I was re-reading...

I still can't understand the *subdomain* thing. Is that what you input in `nextcloudpi-config` in the `letsencrypt` section?

**Comment by Shadowstreik** [comment]
In the LetsEncrypt selection of the NCP Config utility, I entered the address for NCP which includes the sub.

**Comment by Shadowstreik** [comment]
Performed a full rebuild Activated Fail2Ban. Result:
![screenshot from 2017-07-08 13-41-12](https://user-images.githubusercontent.com/29790545/27987919-1702577c-63e4-11e7-88e4-75a869e66650.png)

Disabled Fail2Ban. Result:
![screenshot from 2017-07-08 13-46-03](https://user-images.githubusercontent.com/29790545/27987923-299bffb4-63e4-11e7-9349-8857fb090b09.png)

Conclusion: Fail2Ban was/is the issue.

**Comment by nachoparker** [comment]
good one

does it happen only from outside? or does it also happen using your local network IP?

Could you have a look at the logs for fail2ban, apache and nextcloud?

I want to reproduce it myself

**Comment by nachoparker** [comment]
finally, I could reproduce it 

thanks

**Comment by nachoparker** [comment]
It seems like I have found the root cause. CSP code does not play well with PHP cache.

I have reported the issue to the server developers ( https://github.com/nextcloud/server/issues/5664 )

The most likely outcome is that I will have to blacklist some files from PHP cache.

Does not seem `fail2ban` related. According to my testing, it can be worked around for now by issuing these commands

```
systemctl stop php7.0-fpm
systemctl stop mysqld
systemctl start php7.0-fpm
systemctl start mysqld
```
or rebooting. The problem is that then, if we try to access through IP again (after we last did through domain) we would be in the same situation.

It is a minor issue though, as the user will most likely access the same way always
